apps/dashboard/src/routes/courses/[id]/lessons/+page.svelte

```diff
@@ -81,6 +87,47 @@
       return index + 1;
     }
   }
+
+  function findFirstIncompleteLesson() {
+    return $lessons.find(
+      (lesson) => lesson.lesson_completion.length === 0 && lesson.is_unlocked === true
+    );
+  }
+
+  onMount(() => {
+    // Update loaded lessons when $lessons store changes
+    lessons.subscribe((value) => {
+      loadedLessons = value;
+      // Check if it's the initial load
+      if (isInitialLoad) {
+        isInitialLoad = false;
+      }
+    });
+  });
+
+  $: {
+    let query = new URLSearchParams($page.url.search);
+    nextQueryKey = query.get('next') || '';
+
+    if (nextQueryKey === 'true' && !isFetching) {
+      console.log('lesson in reactive', $lessons);
+    }
+
+    if ($lessons.length > 0) {
+      let incompleteLesson = findFirstIncompleteLesson();
+
+      if (incompleteLesson) {
+        console.log('incompleteLesson', incompleteLesson);
+
+        if (!isInitialLoad) {
+          goto(`/courses/${data.courseId}/lessons/${incompleteLesson.id}`);
+        }
+      } else {
+        console.log('All lessons are complete.');
+        goto(`/courses/${data.courseId}/lessons`);
+      }
+    }
+  }
```

Comment:
@tunny17 Abstract this logic into a function and it should only accept an array of `lessons` and `nextQueryKey`